### ISSUE TITLE
minor issue: rmf-server in example-deployment crashes on first-time launch

### ISSUE THREAD

**Issue by cnboonhan** [open-rmf/rmf-web#446]
Minor issue to flag, when rmf-server is launched in kubernetes `kubectl apply -f k8s/rmf-server.yaml`, it often crashes the first time with the following error. Kubernetes will restart the pod, which will then work properly.

```
INFO:     Started server process [1]
INFO:     Waiting for application startup.
ERROR:    Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 540, in lifespan
    async for item in self.lifespan_context(app):
  File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 481, in default_lifespan
    await self.startup()
  File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 516, in startup
    await handler()
  File "/usr/local/lib/python3.8/dist-packages/api_server/app.py", line 206, in on_startup
    await Tortoise.init(
  File "/usr/local/lib/python3.8/dist-packages/tortoise/__init__.py", line 593, in init
    await cls._init_connections(connections_config, _create_db)
  File "/usr/local/lib/python3.8/dist-packages/tortoise/__init__.py", line 390, in _init_connections
    await connection.create_connection(with_db=True)
  File "/usr/local/lib/python3.8/dist-packages/tortoise/backends/asyncpg/client.py", line 94, in create_connection
    self._pool = await asyncpg.create_pool(None, password=self.password, **self._template)
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/pool.py", line 407, in _async__init__
    await self._initialize()
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/pool.py", line 435, in _initialize
    await first_ch.connect()
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/pool.py", line 127, in connect
    self._con = await self._pool._get_new_connection()
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/pool.py", line 477, in _get_new_connection
    con = await connection.connect(
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/connection.py", line 1980, in connect
    return await connect_utils._connect(
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/connect_utils.py", line 677, in _connect
    raise last_error
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/connect_utils.py", line 661, in _connect
    con = await _connect_addr(
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/connect_utils.py", line 634, in _connect_addr
    tr, pr = await compat.wait_for(connector, timeout=timeout)
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/compat.py", line 103, in wait_for
    return await asyncio.wait_for(fut, timeout)
  File "/usr/lib/python3.8/asyncio/tasks.py", line 494, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.8/dist-packages/asyncpg/connect_utils.py", line 544, in _create_ssl_connection
    tr, pr = await loop.create_connection(
  File "uvloop/loop.pyx", line 2024, in create_connection
  File "uvloop/loop.pyx", line 2001, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

ERROR:    Application startup failed. Exiting.

```

**Comment by koonpeng** [comment]
probably a race condition when the server tries to connect to the db before it is ready, we can try using some of the ready check logic in reporting to fix this.

**Comment by koonpeng** [comment]
fixed by #462
